src/net_processing.cpp

```diff
@@ -1485,6 +1476,7 @@ bool static ProcessHeadersMessage(CNode *pfrom, CConnman *connman, const std::ve
                 // etc), and not just the duplicate-invalid case.
                 pfrom->fDisconnect = true;
             }
+            MaybePunishNode(pfrom->GetId(), state, false, "invalid header received");
```

Comment:
I believe this can result in a behavior change, where we would DoS-ban peers that send duplicate headers that are invalid.